docs/src/usage.md

```diff
@@ -56,3 +56,65 @@ MOI.set(model, DiffOpt.ForwardObjectiveFunction(), ones(2) ⋅ x)
 DiffOpt.forward_differentiate!(model)
 grad_x = MOI.get.(model, DiffOpt.ForwardVariablePrimal(), x)
 ```
+
+3. To differentiate a general nonlinear program, have to use the API for Parameterized JuMP models. For example, consider the following nonlinear program:
```

Comment:
In forward mode,
what happens in the NLP backend if one tries to Set something other than parameters?

In reverse mode, 
what happens in the NLP backend if one tries to Get something other than parameters?

Are these cases tested?

Reply:
Good point, I have now extended test from Parameter interface for the non POI version.